Assets/MixedRealityToolkit.Examples/Demos/UX/BoundingBox/Scripts/BoundingBoxTest.cs

```diff
@@ -24,10 +24,36 @@ public class BoundingBoxTest : InputSystemGlobalListener, IMixedRealitySpeechHan
         private Vector3 cubePosition = new Vector3(0, 0, 2);
         private BoundingBox bbox;
 
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        protected IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        protected virtual void OnEnable()
+        {
+            InputSystem?.RegisterHandler<IMixedRealitySpeechHandler>(this);
+        }
+
+        protected virtual void OnDisable()
+        {
+            InputSystem.UnregisterHandler<IMixedRealitySpeechHandler>(this);
+        }
+
         // Start is called before the first frame update
-        protected override void Start()
+        protected virtual void Start()
```

Comment:
>Start [](start = 31, length = 5)

One thing to be aware of is that the InputSystemGlobalListener.Start() also did this:

[https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_release/Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalListener.cs#L45](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_release/Assets/MixedRealityToolkit.Services/InputSystem/InputSystemGlobalListener.cs#L45)

Without this it's possible that input system could be not valid that this this point.

BUTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT

If you notice something interesting, this override was actually not marked async, so TECHNICALLY there was always a bug here where it wouldn't wait for the base class's await, so actually probably this is just fine.

Just wanted to call this out this as FYI, not because I think it's actually a real thing.

Reply:
We also had a huge number of other components which didn't do registration in 'Start'. I think it is important in core components, which are added by mrtk itself, but shouldn't matter at all for user components.

---
In reply to: [295369799](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4970#discussion_r295369799) [](ancestors = 295369799)